src/test/java/com/example/gradecalculator/service/SubjectServiceTest.java

```diff
@@ -247,6 +247,7 @@ public void testSaveSubjects_WithErrors() {
     public void testDeleteSubject() {
         // Arrange
         when(userSubjectRepository.findById(anyLong())).thenReturn(Optional.of(userSubject));
+        when(userGradeRepository.existsByUserSubjectIdAndUserId(anyLong(),anyLong())).thenReturn(false);
```

Comment:
All new tests missing - there were just some deleted?

Reply:
See what david wrote - you need to test new functionality, not just delete/adjust tests